algorithms/integration/processor.py

```diff
@@ -792,43 +817,111 @@ def compute_processors(self):
         if self.params.mp.method == "multiprocessing" and self.params.mp.nproc > 1:
 
             # Get the maximum shoebox memory
-            max_memory = flex.max(
+            memory_required_per_process = flex.max(
                 self.jobs.shoebox_memory(self.reflections, self.params.shoebox.flatten)
             )
+            logger.debug(
+                "Memory required per process: %.1f GB",
+                memory_required_per_process / 1e9,
+            )
 
             # Compute expected memory usage and warn if not enough
-            total_memory = psutil.virtual_memory().total
-            assert (
-                total_memory is not None and total_memory > 0
-            ), "psutil call appears to have given unexpected output"
-            limit_memory = total_memory * self.params.block.max_memory_usage
-            njobs = int(math.floor(limit_memory / max_memory))
-            if njobs < 1:
-
-                xsize, ysize, zsize = _average_bbox_size(self.reflections)
-                raise RuntimeError(
-                    """
+            available_memory = psutil.virtual_memory().available
```

Comment:
Seeing the same code block repeated - wonder if we should have some dials.util.memory wrapper which computes available wired and virtual memory? Seems like probably useful elsewhere.

Reply:
code duplication resolved by removing task check